src/create-dom-event.ts

```diff
@@ -0,0 +1,104 @@
+import eventTypes from 'dom-event-types'
+
+const keyCodesByKeyName = {
+  backspace: 8,
+  tab: 9,
+  enter: 13,
+  esc: 27,
+  space: 32,
+  pageup: 33,
+  pagedown: 34,
+  end: 35,
+  home: 36,
+  left: 37,
+  up: 38,
+  right: 39,
+  down: 40,
+  insert: 45,
+  delete: 46
+}
+
+function getEventProperties(eventParams) {
+  const { modifier, meta, options } = eventParams
+  const keyCode =
+    keyCodesByKeyName[modifier] ||
+    options.key ||
+    options.keyCode ||
+    options.code
+
+  return {
+    ...options, // What the user passed in as the second argument to #trigger
+    bubbles: meta.bubbles,
+    meta: meta.cancelable,
+    // Any derived options should go here
+    key: keyCode,
+    keyCode,
+    code: keyCode
+  }
+}
+
+function createEvent(eventParams) {
+  const { eventType, meta } = eventParams
+  const metaEventInterface = window[meta.eventInterface]
+
+  const SupportedEventInterface =
+    typeof metaEventInterface === 'function' ? metaEventInterface : window.Event
+
+  const eventProperties = getEventProperties(eventParams)
+
+  const event = new SupportedEventInterface(
+    eventType,
+    // event properties can only be added when the event is instantiated
+    // custom properties must be added after the event has been instantiated
+    eventProperties
+  )
+
+  return event
+}
+
+function createEventForOldBrowsers(eventParams) {
+  const { eventType, modifier, meta } = eventParams
+  const { bubbles, cancelable } = meta
+
+  const event = document.createEvent('Event')
+  event.initEvent(eventType, bubbles, cancelable)
+  event['keyCode'] = keyCodesByKeyName[modifier]
+  return event
+}
+
+export default function createDOMEvent(
+  eventString: String,
+  options: Object = {}
```

Comment:
cool, you can type this like so:

```
interface TriggerOptions {
  key: 'enter' | 'keyup' | ...
}
```

if you aren't sure about TS i can look into this

Reply:
Thanks, I'm not sure I know what you mean. If it's something simple and you want to take care of it yourself that's fine with me, go ahead with it 👍.

Reply:
I added some types, I not sure if that's what you meant.